pkg/operator2/configobservation/configobservercontroller/observe_config_controller.go

```diff
@@ -0,0 +1,82 @@
+package configobservercontroller
+
+import (
+	"k8s.io/client-go/tools/cache"
+
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/configobserver/apiserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation/console"
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation/infrastructure"
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation/routersecret"
+)
+
+type ConfigObserver struct {
+	*configobserver.ConfigObserver
+}
+
+func NewConfigObserver(
+	operatorClient v1helpers.OperatorClient,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	configInformer configinformers.SharedInformerFactory,
+	resourceSyncer resourcesynccontroller.ResourceSyncer,
+	eventRecorder events.Recorder,
+) *ConfigObserver {
+	interestingNamespaces := []string{
+		"openshift-authentication",
+	}
+
+	preRunCacheSynced := []cache.InformerSynced{}
+	for _, ns := range interestingNamespaces {
+		preRunCacheSynced = append(preRunCacheSynced,
+			kubeInformersForNamespaces.InformersFor(ns).Core().V1().ConfigMaps().Informer().HasSynced,
+			kubeInformersForNamespaces.InformersFor(ns).Core().V1().Secrets().Informer().HasSynced,
+		)
+	}
+
+	c := &ConfigObserver{
+		ConfigObserver: configobserver.NewConfigObserver(
+			operatorClient,
+			eventRecorder,
+			configobservation.Listers{
+				SecretsLister: kubeInformersForNamespaces.SecretLister(),
+
+				APIServerLister_:     configInformer.Config().V1().APIServers().Lister(),
+				ConsoleLister:        configInformer.Config().V1().Consoles().Lister(),
+				InfrastructureLister: configInformer.Config().V1().Infrastructures().Lister(),
+				ResourceSync:         resourceSyncer,
+				PreRunCachesSynced: append(preRunCacheSynced,
+					operatorClient.Informer().HasSynced,
+
+					configInformer.Config().V1().APIServers().Informer().HasSynced,
+					configInformer.Config().V1().Consoles().Informer().HasSynced,
+					configInformer.Config().V1().Infrastructures().Informer().HasSynced,
+				),
+			},
+			apiserver.ObserveAdditionalCORSAllowedOrigins,
+			apiserver.ObserveTLSSecurityProfile,
```

Comment:
@stlaz I think that we have to change it because we need separate settings for each operand.

Reply:
they now observe separately into `oauthServer` and `oauthAPIServer` field in the `observedConfig`